index.js

```diff
@@ -318,6 +318,7 @@ module.exports = function (filename, opts) {
       blockBuf.length,
       blockIndex * blockSize
     )
+    writingBlockIndex = blockIndex
```

Comment:
wow, this looks like it was a bug before

Reply:
Well, now that you raised this topic, what if a `write` takes longer than `writeTimeout` and then a second `write()` would start before the first `write()` gets the chance to set `writingBlockIndex = -1`? Would there be problems? Like in `onDrain` where we read `writingBlockIndex`

Reply:
@arj03 in case you missed this